models/repo.go

```diff
@@ -1583,14 +1584,24 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 	}
 
 	repos = make([]*Repository, 0, opts.PageSize)
-
-	// Append conditions
-	sess := x.Where("LOWER(lower_name) LIKE ?", "%"+opts.Keyword+"%")
-	if opts.OwnerID > 0 {
-		sess.And("owner_id = ?", opts.OwnerID)
+	sess := x.Alias("r")
+	// Attempt to find repositories that opts.UserId has access to
+	// This does not include other people's private repositories even if opts.UserId is an admin
+	if !opts.Private && opts.UserID > 0 {
+		sess.Join("LEFT", []string{"team_repo", "tr"}, "tr.repo_id=r.id")
```

Comment:
Use `access` table to get not-owned, private and has access to repositories is needed, here you didn't consider the case where user is collaborator of private repositories.

So what we to do:

1. To include repos user directly owned
2. To find private repos that has record in `access` table (has such record means user has at least read permission)